aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3Decorator.kt

```diff
@@ -209,21 +205,6 @@ class FilterEndpointTests(
         }
 }
 
-// TODO(P96049742): This model transform may need to change depending on if and how the S3 model is updated.
-private class AddOptionalAuth {
```

Comment:
Why are we removing this? What was our plan again to keep s3 working without auth (since we shipped it this way)?

Reply:
Will definitely come back to this and assess the risk of removing the transformer.

Reply:
We should _not_ conflate `no_credentials` with this. All that `no_credentials` does is set the credential provider to explicitly unset. S3 works with no credentials due to the previous customization, **not** because `no_credentials` was called.